File: src/main/java/um/haberes/core/repository/ClaseRepository.java
package um.haberes.core.repository;

import org.springframework.data.jpa.repository.JpaRepository;
import um.haberes.core.kotlin.model.Clase;

import java.util.Optional;

public interface ClaseRepository extends JpaRepository<Clase, Integer> {

    Optional<Clase> findTopByOrderByClaseIdDesc();
    Optional<Clase> findByClaseId(Integer claseId);

}

File: src/main/java/um/haberes/core/kotlin/repository/ClaseRepository.kt
package um.haberes.core.kotlin.repository

import org.springframework.data.jpa.repository.JpaRepository
import um.haberes.core.kotlin.model.Clase
import java.util.Optional

interface ClaseRepository : JpaRepository<Clase?, Int?> {

    fun findTopByOrderByClaseIdDesc(): Optional<Clase?>?

    fun findByClaseId(claseId: Int?): Optional<Clase?>?
}
